MaxText/input_pipeline/input_pipeline_interface.py

```diff
@@ -188,7 +186,7 @@ def make_mixed_train_iterator_and_tokenizer(config, mesh, add_bos, add_eos):
       print("Overwrite both add_bos and add_eos to False")
       return make_c4_mlperf_train_iterator_and_tokenizer(config, mesh, add_bos=False, add_eos=False, process_indices = process_indices)
   else:
-    return BadSyntheticDataIterator(config, mesh), None, get_tokenizer(config.tokenizer_path, add_bos, add_eos)
+    return BadSyntheticDataIterator(config, mesh).get_iterator(), None, get_tokenizer(config.tokenizer_path, add_bos, add_eos)
```

Comment:
`BadSyntheticDataIterator` shouldn't have a get_iterator? the names don't make sense. I think `BadSyntheticDataIterator` should return an iterator or if you must you could have a `BadSyntheticDataIteratorFactory` (but why?)